factory/README.md

```diff
@@ -0,0 +1,172 @@
+# cypress/factory
+
+cypress/factory is a docker container that can be used with `args` to generate a docker container with specific versions of:
+
+* node
+* yarn
+* chrome
+* firefox
+* edge
+* cypress
+
+## Benefits
+
+* Freedom to choose which versions to test against.
+* No need to wait on an official release to test the latest version of a browser.
+* Smaller docker sizes especially when not including unused browsers.
+* Easily test multiple browser versions.
+* Reduced maintenance and pull requests for the cypress-docker repo.
+* Ability to offer more variations of docker containers at low cost.
+
+## Usage
+
+The cypress docker factory works by relying on the `ONBUILD` docker command to run commands at build time of the the consuming container. `Args` can be specified in a number of ways, some of which are demonstrated below. For each of these examples we are building the equivalent of the `cypress/browsers` docker container. In each instance since the example is only testing the chrome version, the examples could just install chrome by itself if the other browsers were not used.
+
+### In the dockerfile
+
+Args can be defined directly in the dockerfile.
+
+dockerfile
+
+```dockerfile
+# Args are defined in the dockerfile before the FROM command.
+# Using these args will cause an image to be created with node, chrome, firefox and edge.
+ARG CHROME_VERSION='107.0.5304.121-1'
+ARG EDGE_VERSION='100.0.1185.29-1'
+ARG FIREFOX_VERSION='107.0'
+
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+build commands
+
+```bash
+docker build . -t test
+docker run -it test npm run test -b chrome
+```
+
+### At build time
+
+Args can be passed to the docker build command.
+
+dockerfile
+
+```dockerfile
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+build commands
+
+```bash
+docker build . --build-arg CHROME_VERSION='107.0.5304.121-1' --build-arg EDGE_VERSION='100.0.1185.29-1' --build-arg FIREFOX_VERSION='107.0' -t test
+
+docker run test npm run test -b chrome
+```
+
+### In docker-compose.yml
+
+Finally, args can be specified in the docker-compose.yml file.
+
+docker-compose.yml
+
+```yml
+version: '3'
+
+services:
+  test:
+    args:
+      CHROME_VERSION: '107.0.5304.121-1'
+      EDGE_VERSION: '100.0.1185.29-1'
+      FIREFOX_VERSION: '107.0'
+    build:
+      context: .
+    command: npm run test
+```
+
+dockerfile
+
+```dockerfile
+FROM cypress/factory
+
+COPY . /opt/app
+WORKDIR /opt/app
+RUN npm install --save-dev cypress
+```
+
+docker commands
+
+```bash
+docker-compose build test
+
+docker-compose run test
+```
+
```

Comment:
I'd add the example about only running chrome and you could build a slimmer browser image here

Reply:
added